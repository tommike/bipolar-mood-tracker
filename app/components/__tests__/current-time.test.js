import React from 'react';
import { shallow } from 'enzyme';
import CurrentTime from '../current-time';

jest.mock('moment', () => () => ({
  format: args => {
    if (args === 'MMM Do YYYY') {
      return 'Oct 8th 2018';
    }
    return '2018-10-08T17:27:08+02:00';
  },
}));

let wrapper;
const createWrapper = () => shallow(<CurrentTime />);

describe('<CurrentTime />', () => {
  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('renders without throwing an error', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct dateTime and displayTime to HTML', () => {
    expect(wrapper.find('time').props().dateTime).toBe('2018-10-08T17:27:08+02:00');
    expect(wrapper.find('time').text()).toBe('Oct 8th 2018');
  });

  // it('calls handleVisibilityChange when visibilitychange event is triggered') {
});
