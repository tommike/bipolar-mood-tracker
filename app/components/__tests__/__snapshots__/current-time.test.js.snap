// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentTime /> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentTime />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <time
        className="date__datetime"
        dateTime="2018-10-08T17:27:08+02:00"
      >
        Oct 8th 2018
      </time>,
      "className": "date",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Oct 8th 2018",
        "className": "date__datetime",
        "dateTime": "2018-10-08T17:27:08+02:00",
      },
      "ref": null,
      "rendered": "Oct 8th 2018",
      "type": "time",
    },
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <time
          className="date__datetime"
          dateTime="2018-10-08T17:27:08+02:00"
        >
          Oct 8th 2018
        </time>,
        "className": "date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oct 8th 2018",
          "className": "date__datetime",
          "dateTime": "2018-10-08T17:27:08+02:00",
        },
        "ref": null,
        "rendered": "Oct 8th 2018",
        "type": "time",
      },
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
